client/verta/verta/pipeline/_pipeline_graph.py

```diff
@@ -82,15 +88,17 @@ def _validate_steps(self, steps: Set[PipelineStep]) -> Set[PipelineStep]:
         TypeError
             If steps is not a set of PipelineStep objects.
         """
-        if not isinstance(steps, set):
-            raise TypeError(f"steps must be type set, not {type(steps)}")
+        if not isinstance(steps, (list, set, tuple)):
+            raise TypeError(
+                f"steps must be type list, set, or tuple, not {type(steps)}"
+            )
         for step in steps:
             if not isinstance(step, PipelineStep):
                 raise TypeError(
                     f"individual steps of a PipelineGraph must be type"
                     f" PipelineStep, not {type(step)}."
                 )
-        return steps
+        return set(steps)
```

Comment:
I'd strongly recommend having a validation method _only_ do validation, and not have it perform type casting.

Reply:
Ok, cool.  We can just do this in the "set" method rather than the validator method